### PR TITLE
feat: edit file form UI

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: #2249

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

https://deploy-preview-2257--regen-storybook.netlify.app/?path=/story/forms-editfileform--basic

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-components/package.json [hunk 1]
```diff
@@ -64,7 +68,7 @@
     "react-helmet": "^6.0.0",
     "react-image-crop": "^9.0.5",
     "react-lazyload": "^3.2.0",
-    "react-map-gl": "5.3.21",
+    "react-map-gl": "^7.1.7",
```
Comment (blushi) [inline web-components/package.json:71]: upgraded our react-map-gl version so there are a few changes below that can be found in the existing `StaticMap` component that are not necessarily related to the initial purpose of this work

**Comment by blushi** [comment]
@erikalogie you can test this on https://deploy-preview-2257--regen-storybook.netlify.app/?path=/story/forms-editfileform--basic

While testing, please be aware that:
- the inputs font size will be updated in another PR: https://github.com/regen-network/regen-web/pull/2251 (ref: https://github.com/regen-network/regen-web/pull/2251#discussion_r1444562210)
- the reverse geocoding for getting the place name for a particular point on the map and displaying it in parenthesis is not enabled here, since it requires to make API calls to mapbox (which are limited) and I thought it's best to skip it in storybook but everything is in place to wire it up in the app later on.

**Comment by erikalogie** [comment]
@blushi this looks great! I would just make it so that the location input for "choose a certain location on the map" is floating above the map with a little bit of space around it, like in the comps. I think that makes it look more like something you can interact with than the current version.

Otherwise, great work

**Comment by blushi** [comment]
> @blushi this looks great! I would just make it so that the location input for "choose a certain location on the map" is floating above the map with a little bit of space around it, like in the comps. I think that makes it look more like something you can interact with than the current version.
> 
> Otherwise, great work

That's fixed now!

**Comment by erikalogie** [comment]
Perfect, this all looks good to me now

**Comment by aaronc** [comment]
One thing I find a bit unusual is that I must drag the map layer behind the pin instead of the pin itself to choose a custom location. My first instinct was to move the pin but nothing happened.

**Comment by blushi** [comment]
> One thing I find a bit unusual is that I must drag the map layer behind the pin instead of the pin itself to choose a custom location. My first instinct was to move the pin but nothing happened.

On figma, there was a specific dev note about that, which is why I've implemented it this way: https://www.figma.com/file/Bksz1JeDYxQVIXdI46EgPT/Project-Posts?node-id=1410%3A79803&mode=dev
cc/ @erikalogie

**Comment by erikalogie** [comment]
> > One thing I find a bit unusual is that I must drag the map layer behind the pin instead of the pin itself to choose a custom location. My first instinct was to move the pin but nothing happened.
> 
> On figma, there was a specific dev note about that, which is why I've implemented it this way: https://www.figma.com/file/Bksz1JeDYxQVIXdI46EgPT/Project-Posts?node-id=1410%3A79803&mode=dev cc/ @erikalogie
Yes this was intentional. I think it is a more elegant interaction for adding a single point to a map, and the only way to do it properly on mobile. If we had multiple points that needed to be added then dragging them around might make sense on desktop, but this is much easier for the user even if it isn't immediately apparent, I think it is very quickly discoverable. 
> 
> One benefit is if you want to put the pin outside the current map view, you don't have to first drag the map around and then move the pin like here: https://visgl.github.io/react-map-gl/examples/draggable-markers, you just need to drag the map layer.
Yes exactly. Also if there is a compelling reason to have it be draggable, then we'd need to retain this existing functionality for mobile and just enable it on desktop.

**Comment by erikalogie** [comment]
> Sounds good 👍. It's a small feature that might not even get used that much. Was just noticing it didn't work as I initially expected. Maybe we could just add a tiny bit of help text to drag the map itself?

Ok, maybe like this? @aaronc 
Screen Shot 2024-01-22 at 10 35 29 AM

**Pull request merged**